kuksa-client/kuksa_client/grpc/aio.py

```diff
@@ -274,6 +286,7 @@ async def subscribe(self, *,
             rpc_kwargs
                 grpc.*MultiCallable kwargs e.g. timeout, metadata, credentials.
         """
+        rpc_kwargs.__setitem__("metadata", self.rpc_metadata)
```

Comment:
This is still overwriting the "metadata" key in `rpc_kwargs` just like before.

Maybe I need to expand on what I mean here.

Ok, so I don't know if you're familiar with the `*args` & `**kwargs` concept in python. Anyway, what the `*` and `**` means in practice is that this is a way to pack the arguments that are passed into a function in python into a single argument. A single `*` refers to packed positional arguments, and a double `**` refers to the packed keyword arguments. The same syntax is also used to unpack variables into function arguments when calling a function. But let's start with the packing.

In this case we have the following function exposed in the library:
```
async def subscribe(self, *, entries: Iterable[SubscribeEntry], **rpc_kwargs)
```

What this means, is that if someone calls this function and want to supply some extra metadata, for example to enable connecting to databroker using a DAPR gRPC proxy. they would call it like this:
```
client.subscribe(entries=[entry], metadata=[("dapr-app-id", "server")])
```

Now, how does this work? `metadata` isn't an argument in the function. That's where pythons argument packing comes into play. Inside the `subscribe` function `**rpc_kwargs` will contain the packed _keyword arguments_ like this:

```
{
  "metadata": [("dapr-app-id", "server")]
}
```

which will subsequently be supplied to the actual call to the stub like this:
```
resp_stream = self.client_stub.Subscribe(req, **rpc_kwargs)
```

which is the same as calling it like this:

```
resp_stream = self.client_stub.Subscribe(req, metadata=[("dapr-app-id", "server")])
```

So this is what I mean when I say that you shouldn't overwrite "metadata" if it's already present in `**rpc_kwargs`. _If_ "metadata" is already present, another key, "authorization" should be added to it without overwriting "dapr-app-id" (taking the example provided above).